### PR TITLE
Update SettingsScreen

### PR DESCRIPTION
**Pull request by nada-ezziti**
### Pull Request : Update SettingsScreen

This PR introduces updates to `SettingsScreen` and related files to align with our Figma design specifications, achieve code optimization, and add new user-facing functionality. Key changes include:

1. **Design and Code Improvements**: Updated `SettingsScreen` to match the Figma design (color scheme and layout), reducing code duplication by implementing the common back button component.

2. **New Feature - Update Username**: Added a feature allowing users to change their username directly from `SettingsScreen`. This functionality leverages two new functions, `getUserName` and `updateUserName`, which are tested with unit tests to ensure data integrity and seamless integration with Firestore for real-time updates.

3. **Test Coverage**: `SettingsScreen` now has 92% test coverage, covering new and existing features.

4. **MainActivity Adaptation**: Adjusted `MainActivity` to accommodate `SettingsScreen` changes by reducing the number of arguments required.

Capture d’écran 2024-10-31 à 12 29 16

### CODE REVIEW

**Review by ThomasMPont — CHANGES_REQUESTED** [review]
This branch raises [2 new issues on SonarCloud](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&pullRequest=151&id=Signify-epfl_signify-app). They look rather simple to fix. Could you take a look at them?

**Review by ThomasMPont — DISMISSED** [review]

File: app/src/androidTest/java/com/github/se/signify/ui/screens/MainActivityTest.kt
Comment (ziyadbagh) [inline app/src/androidTest/java/com/github/se/signify/ui/screens/MainActivityTest.kt:1]: Good Job !! 92% is a very good coverage

File: app/src/androidTest/java/com/github/se/signify/ui/screens/profile/SettingsScreenTest.kt
Comment (ziyadbagh) [inline app/src/androidTest/java/com/github/se/signify/ui/screens/profile/SettingsScreenTest.kt:1]: Good Job !! 92% is a very good coverage

**Review by ThomasMPont — APPROVED** [review]
Good job on not having any issues or duplications on SonarCloud.
The logic of the screen with Firebase seems to be working correctly. And the screen navigates fine.

The screen's UI feels pretty unintuitive and clunky, and will definitely need to be improved along with its Figma design in the future.
I've had to push a change to make the screen scrollable to ensure it is usable on any device. The screen main Composable should be further changed to try to fit all its elements on screen when possible.

Since the bulk of the work for this PR concerns backend logic, which is working well, I think we would still benefit from merging it right now.

**Review by ziyadbagh — APPROVED** [review]

**Pull request merged**